fcrepo4/fcrepo4
# Make Repository rebuilder handle binaries

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3268

# What does this Pull Request do?
Assigns the binary as the parent of the description when sending information to the persistence layer and then uses the parent and the interaction model to ensure the binary description gets re-added to the index correctly.

# How should this be tested?

Before PR.

1. Add a binary to Fedora. 
     ```
     > curl -ufedoraAdmin:fedoraAdmin -i -H"Content-type: text/plain" -H"Slug: binary1" --data-binary "Some text content" -XPOST http://localhost:8080/rest
     ```
1. `GET` the binary and its metadata, validate it works.
    ```
    > curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/binary1
    > curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/binary1/fcr:metadata
     ```
1. Shutdown Fedora and delete the index file, `fedoraToOcflIndex.tsv` in the OCFL work directory.
1. Restart Fedora
1. Try to get the binary and metadata, see errors
1. Shutdown Fedora, pull in this PR, compile
1. Delete the index file again
1. Restart Fedora
1. Try to get the binary and metadata, see correct results.

# Interested parties
@fcrepo4/committers
